src/ThreatMonitoring/ConfigClient.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace UKFast\SDK\ThreatMonitoring;
+
+use UKFast\SDK\ThreatMonitoring\Entities\Config;
+
+class ConfigClient extends Client
+{
+    const FIM_DIRECTORIES_MAP = [
+        'check_all' => 'checkAll',
+        'check_attrs' => 'checkAttrs',
+        'check_group' => 'checkGroup',
+        'check_inode' => 'checkInode',
+        'check_md5sum' => 'checkMd5sum',
+        'check_mtime' => 'checkMtime',
+        'check_owner' => 'checkOwner',
+        'check_perm' => 'checkPerm',
+        'check_sha1sum' => 'checkSha1sum',
+        'check_sha256sum' => 'checkSha256sum',
+        'check_size' => 'checkSize',
+        'check_sum' => 'checkSum',
+        'follow_symbolic_link' => 'followSymbolicLink',
+        'recursion_level' => 'recursionLevel',
+        'report_changes' => 'reportChanges'
+    ];
+
+    const LOG_MAP = [
+        'log_format' => 'logFormat'
+    ];
+
+    /**
+     * Get the config data by the group name
+     * @param $groupName
+     * @return Config
+     */
+    public function getByGroupName($groupName)
+    {
+        $response = $this->get('v1/configs/groups/' . $groupName);
+        $body = $this->decodeJson($response->getBody()->getContents());
+
+        return $this->serializeResponse($body->data);
+    }
+
+    /**
+     * Get the config data by the agent id
+     * @param $agentId
+     * @return Config
+     */
+    public function getByAgentId($agentId)
+    {
+        $response = $this->get('v1/configs/agents/' . $agentId);
+        $body = $this->decodeJson($response->getBody()->getContents());
+
+        return $this->serializeResponse($body->data);
+    }
+
+    /**
+     * Serialize the response data
+     * @param $data
+     * @return Config
+     */
+    public function serializeResponse($data)
+    {
```

Comment:
One thing I'd suggest here is to make entities for your logs.

As a consumer, it is nicer to use when all I have to do is chain arrows(e.g. `$config->logs[0]->editable`)  instead of an arrow chain followed by array syntax (e.g. `$config->logs[0]->editable`).

This is used in quite a few other places across the SDK, e.g. https://github.com/ukfast/sdk-php/blob/master/src/DDoSX/AccessLogClient.php#L126 & https://github.com/ukfast/sdk-php/blob/master/src/PSS/RequestClient.php#L157

Reply:
This would also help with IDE hints.

One thing not to forget is to use `syncOriginalAttributes()` after doing that, otherwise your fresh entity weill be marked as "dirty"

Reply:
I'll have a look at this now, thabks Phil

Reply:
I have updates the response, can you also check that the entities are now correct. Thanks